src/components/pages/user/userPosts/index.tsx

```diff
@@ -1,14 +1,16 @@
 import Loader from "@/app/loading";
 import { PostInstanceType } from "@/types";
 import Image from "next/image";
-import { Suspense, useState } from "react";
+import Link from "next/link";
+import { Suspense } from "react";
 import { Bookmark, Download, Heart, MessageCircle, Share } from "react-feather";
 import { useSelector } from "react-redux";
 
 type FormatOnType = "seconds" | "minutes" | "hours" | "days";
+
 interface UserPostsProps {
   userId: string;
-  userName: string;
+  userName: string | undefined;
```

Comment:
Do we actually need to make userName undefined since it is 100% guranteed that it is stored in user collection